contracts/modules/staking/StakingNFTDescriptor.sol

```diff
@@ -0,0 +1,135 @@
+        "data:application/json;base64,",
+        Base64.encode(
+          bytes(
+            abi.encodePacked(
+              '{"name":"', params.name, '",',
+              '"description":"', description, '",',
+              '"image": "', "data:image/svg+xml;base64,", image,
+              '"}'
+            )
+          )
+        )
+      )
+    );
+  }
+
+  function buildDescription(StakingTokenURIParams calldata params) public view returns (string memory description) {
+    // Check if token exists
+    if (params.poolId != 0 && params.owner != address(0)) {
+      (string memory depositInfo, uint totalStake, uint pendingRewards) = getActiveDeposits(params);
+
+      // Add pool info
+      description = append("This NFT represents a deposit into staking pool: ", uint(uint160(params.stakingPool)).toHexString());
+      description = appendWithNewline(description, "Pool ID: ", params.poolId.toString());
+
+      // No active deposits, assume it has expired
+      if (totalStake == 0) {
+        description = appendWithNewline(description, "Deposit has expired!");
+        return description;
+      }
+
+      // Add deposit info
+      description = appendWithNewline(description, "Staked amount: ", totalStake.toString(), " NXM");
+      description = appendWithNewline(description, "Pending rewards: ", pendingRewards.toString(), " NXM");
+      description = appendWithNewline(description, "Active deposits: ", depositInfo);
+      return description;
+    }
+    description = string(abi.encodePacked("Token id ", params.tokenId.toString(), " is not minted"));
+  }
+
+
+  function getActiveDeposits(StakingTokenURIParams calldata params)
+  public
+  view
+  returns (string memory depositInfo, uint totalStake, uint pendingRewards) {
+    IStakingPool stakingPool = IStakingPool(params.stakingPool);
+    uint256 activeStake = stakingPool.getActiveStake();
+    uint256 stakeSharesSupply = stakingPool.getStakeSharesSupply();
+
+    // Get total stake from each active tranche
+    for (uint256 i = 0; i < MAX_ACTIVE_TRANCHES; i++) {
+      // get deposit
+      (uint256 lastAccNxmPerRewardShare, uint256 _pendingRewards, uint256 _stakeShares, uint256 _rewardsShares) =
+      stakingPool.getDeposit(params.tokenId, (block.timestamp / TRANCHE_DURATION) + i);
+
+      // no active stake, skip this tranche
+      if (_rewardsShares == 0) {
+        continue;
+      }
+
+      // update pending rewards
+      uint newRewardPerShare = stakingPool.getAccNxmPerRewardsShare() - lastAccNxmPerRewardShare;
+      pendingRewards += _pendingRewards + (newRewardPerShare * _rewardsShares) / ONE_NXM;
+
+      // update total stake
+      uint stake = (activeStake * _stakeShares) / stakeSharesSupply;
+      totalStake += stake;
+
+      // calculate days left until stake expires
+      uint daysLeftInTranche = (TRANCHE_DURATION - (block.timestamp % TRANCHE_DURATION)) / 1 days;
+      uint daysLeft = daysLeftInTranche + (i * TRANCHE_DURATION / 1 days);
```

Comment:
Should I turn this into DD/MM/YYYY format?

Reply:
Should probably show it as expiration date, yes. Easier to grasp.